### ISSUE TITLE
`CouldBeSequence` false negative when adding operations after `toList()`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#8545]
## Expected Behavior
```kotlin
            val bar = listOf<String?>("bar")
                .asSequence()
                .filterIsInstance<String>()
                .filterNot { true }
                .filterNotNull()
                .take(1)
                .drop(1)
                .takeWhile { true }
                .toList()
                .takeLast(2)
                .mapNotNull { it }
```

Should throw an issue because there is not a good reason to don't move the `toList()` at the end of the chain.

## Observed Behavior
This snipped doesn't throw any issue

## Context
For more context you can read https://github.com/detekt/detekt/pull/8200#discussion_r2301622748

## Your Environment
* Version of detekt used: main

**Comment by atulgpt** [comment]
not stale
